scripts/partition_manager.py

```diff
@@ -0,0 +1,353 @@
+            if parent_partition in reqs:
+                if not first_parent_partition:
+                    first_parent_partition = parent_partition
+                size += reqs[parent_partition]['size']
+        if size == 0:
+            raise RuntimeError("No compatible parent partition found for %s" % sp_name)
+        size = size // len(sp_values['sub_partitions'])
+        address = reqs[first_parent_partition]['address']
+        for sub_partition in sp_values['sub_partitions']:
+            sp_key_name = "%s_%s" % (sp_name, sub_partition)
+            reqs[sp_key_name] = dict()
+            reqs[sp_key_name]['size'] = size
+            reqs[sp_key_name]['address'] = address
+            address += size
+
+
+def write_override_files(adr_map):
+    for img, conf in adr_map.items():
+        if 'out_path' not in conf.keys() or 'out_dir' not in conf.keys():
+            continue  # The 'image' being inspected is not an executable, just a place-holder.
+        open(conf['out_path'], 'w').write('''\
+#undef CONFIG_FLASH_BASE_ADDRESS
+#define CONFIG_FLASH_BASE_ADDRESS %s
+#undef CONFIG_FLASH_LOAD_OFFSET
+#define CONFIG_FLASH_LOAD_OFFSET 0
+#undef CONFIG_FLASH_LOAD_SIZE
+#define CONFIG_FLASH_LOAD_SIZE %s
+''' % (hex(conf['address']), hex(conf['size'])))
+
+
+def get_flash_size(config):
+    config.seek(0)  # Ensure that we search the entire file
+    for line in config.readlines():
+        match = re.match(r'#define CONFIG_FLASH_SIZE (\d*)', line)
+        if match:
+            return int(match.group(1)) * 1024
+    raise RuntimeError("Unable to find 'CONFIG_FLASH_SIZE' in any of: %s" % config.name)
+
+
+def generate_override(input_files, output_file_name, configs, app_override_file):
+    adr_map = dict()
+    load_adr_map(adr_map, input_files, output_file_name, app_override_file)
+    load_size_config(adr_map, configs)
+    flash_size = get_flash_size(configs[0])
+    solution, sub_partitions = resolve(adr_map)
+    set_addresses(adr_map, solution, flash_size)
+    set_sub_partition_address_and_size(adr_map, sub_partitions)
+    return adr_map
+
+
+def get_header_guard_start(filename):
+    macro_name = filename.split('.h')[0]
+    return '''/* File generated by %s, do not modify */
+#ifndef %s_H__
+#define %s_H__''' % (__file__, macro_name.upper(), macro_name.upper())
+
+
+def get_header_guard_end(filename):
+    return "#endif /* %s_H__ */" % filename.split('.h')[0].upper()
+
+
+def write_pm_config(adr_map, pm_config_file):
+    lines = list()
+    lines.append(get_header_guard_start(pm_config_file))
+    flash_area_id = 0
+
+    lines.append("\n/* Indirect, iterable list of flash areas */")
+    for area_name, area_props in sorted(adr_map.items(), key=lambda key_value: key_value[1]['address']):
+        lines.append("#define PM_%d_LABEL %s" % (flash_area_id, area_name.upper()))
+        adr_map[area_name]['flash_area_id'] = flash_area_id
+        flash_area_id += 1
+
+    for area_name, area_props in adr_map.items():
```

Comment:
How come the other two are sorted, but not this one?

Reply:
IMO these should be sorted by ID.